src/environmentd/tests/sql.rs

```diff
@@ -2548,19 +2552,19 @@ fn test_query_on_dropped_source() {
             .connect(postgres::NoTls)
             .unwrap();
         let query_thread = std::thread::spawn(move || {
-            let res = query_client.batch_execute(query);
+            let res = query_client.batch_execute(subscribe);
             assertion(res, rx);
         });
 
         Retry::default()
             .max_duration(Duration::from_secs(10))
             .retry(|_| {
                 let row = client
-                    .query_one("SELECT count(*) FROM mz_internal.mz_compute_exports;", &[])
+                    .query_one("SELECT count(*) FROM mz_internal.mz_subscriptions;", &[])
```

Comment:
`mz_compute_exports` is updated asynchronously so the heavy the load on the Coord the slower it will take. `mz_subscriptions` is updated synchronously so it's guaranteed to happen immediately after the subscribe starts.

That doesn't really answer your question, but I don't have a good answer. Looking into why `mz_compute_exports` is so slow is probably a good idea, but I also think it can be done separately from de-flaking this test since it's just an implementation detail of the test and not what the test was testing directly.